sycl/include/sycl/types.hpp

```diff
@@ -142,20 +144,61 @@ using select_apply_cl_t = std::conditional_t<
 template <typename T> struct vec_helper {
   using RetType = T;
   static constexpr RetType get(T value) { return value; }
+  static constexpr RetType set(T value) { return value; }
```

Comment:
Nit; I am not much for the name "set". What does it set? "get" isn't great for that matter, but it sounds more and more like we are mutating something, which we are not.